engine/cluster/src/main/java/org/opennms/alec/engine/cluster/AbstractClusterEngine.java

```diff
@@ -119,6 +125,21 @@
     // Used to prevent processing callbacks before the init has completed
     private final CountDownLatch initLock = new CountDownLatch(1);
 
+    private final AtomicInteger numVerticesFromLastTick = new AtomicInteger(-1);
+    private final AtomicInteger numEdgesFromLastTick = new AtomicInteger(-1);
+    private final AtomicInteger numAlarmsFromLastTick = new AtomicInteger(-1);
+
+    public AbstractClusterEngine(MetricRegistry metrics) {
+        metrics.gauge(name("vertices"), () -> numVerticesFromLastTick::get);
+        metrics.gauge(name("edges"), () -> numEdgesFromLastTick::get);
+        metrics.gauge(name("alarms"), () -> numAlarmsFromLastTick::get);
+        metrics.gauge(name("situations"), () -> () -> {
```

Comment:
Would be a bit cleaner to create a new method on the AbstractClusterEngine to expose the size of the situations and move the synchronization there.

Then this line can be cleaned up to:

` metrics.gauge(name("situations"), () -> this::getSituationsCount);`

Reply:
Also, not sure when exactly these suppliers are called. But is it possible that these are called during init of this class? Init isn't synchronized and does puts to the situationsById map.

Reply:
The suppliers will be called when the metrics are accessed. In our case, this is when someone access the MBeans.

Moved this to a getter and made `init()` synchronized.